src/server/pure.ts

```diff
@@ -20,8 +20,12 @@ function createServerRenderer<TProps, TResult>(
     render(props?: TProps) {
       renderProps = props
       act(() => {
-        const serverOutput = ReactDOMServer.renderToString(testHarness(props))
-        container.innerHTML = serverOutput
+        try {
+          const serverOutput = ReactDOMServer.renderToString(testHarness(props))
+          container.innerHTML = serverOutput
+        } catch (e: unknown) {
+          rendererProps.setError(e as Error)
+        }
```

Comment:
turns out error boundaries throw synchronously when server rendering.  This is the only renderer that has any special treatment for error handling and only in the initial render function.  The error boundary behaves normally once it has been hydrated.